examples/move/big_vector/sources/big_vector.move

```diff
@@ -0,0 +1,1274 @@
+        let (ref, offset) = self.slice_around(ix);
+        let slice = self.borrow_slice_mut(ref);
+        &mut slice[offset]
+    }
+
+    // === BigVector Mutators ===
+
+    /// Add `val` to `self` at index `key`. Aborts if `key` is already
+    /// present in `self`.
+    public fun insert<E: store>(self: &mut BigVector<E>, key: u128, val: E) {
+        self.length = self.length + 1;
+
+        if (self.root_id == NO_SLICE) {
+            self.root_id = self.alloc(singleton(key, val));
+            return
+        };
+
+        let (root_id, depth) = (self.root_id, self.depth);
+        let (key, other) = self.slice_insert(root_id, depth, key, val);
+
+        if (other != NO_SLICE) {
+            self.root_id = self.alloc(branch(key, root_id, other));
+            self.depth = self.depth + 1;
+        }
+    }
+
+    /// Adds key value pairs from `keys` and `vals` to `self`.
+    /// Requires that `keys` and `vals` have the same length, and that
+    /// `keys` is in sorted order.
+    ///
+    /// Aborts if any of the keys are already present in `self`, or
+    /// the requirements on `keys` and `vals` are not met.
+    public fun insert_batch<E: store>(
+        _self: &mut BigVector<E>,
+        _keys: vector<u128>,
+        _vals: vector<E>,
+    ) {
+        abort 0
+    }
+
+    /// Remove the element with key `key` from `self`, returning its
+    /// value. Aborts if `key` is not found.
+    public fun remove<E: store>(self: &mut BigVector<E>, key: u128): E {
+        self.length = self.length - 1;
+
+        if (self.root_id == NO_SLICE) {
+            abort ENotFound
+        };
+
+        let (root_id, depth) = (self.root_id, self.depth);
+        let (val, rm_fix, _) = self.slice_remove(
+            NO_SLICE,
+            0u128,
+            root_id,
+            0u128,
+            NO_SLICE,
+            depth,
+            key,
+        );
+
+        if (rm_fix == RM_FIX_EMPTY) {
+            if (self.depth == 0) {
+                let Slice<E> {
+                    prev: _,
+                    next: _,
+                    keys: _,
+                    vals,
+                } = df::remove(&mut self.id, root_id);
+
+                // SAFETY: The slice is guaranteed to be empty because
+                // it is a leaf and we received the RM_FIX_EMPTY
+                // fix-up.
+                vals.destroy_empty();
+
+                self.root_id = NO_SLICE;
+            } else {
+                let mut root: Slice<u64> = df::remove(&mut self.id, root_id);
+                self.root_id = root.vals.pop_back();
+                self.depth = self.depth - 1;
+            }
+        };
+
+        val
+    }
+
+    /// Remove the elements between `lo` (inclusive) and `hi`
+    /// (exclusive) from `self`.
+    public fun remove_range<E: store + drop>(
+        _self: &mut BigVector<E>,
+        _lo: u128,
+        _hi: u128,
+    ) {
+        abort 0
```

Comment:
Ditto, out of scope for prototype (useful for market maker workloads).

Reply:
See above, I do not think separate batch insert / delete with aborts is as useful.